### PR TITLE
fix(strike_tests): added better tests for strikes

### LINKED ISSUE #267: Legg til tester for Strikes
Legg til tester for logikken i strikes, altså både model, og integration tester

### PR DESCRIPTION
**Pull request by CJGutz**
## Proposed changes
Added tests for strike. These tests replaces the previous tests which poorly covered strike's functionaility. I have added both integration and model tests. The model tests focuses on expration date of strikes and the integration tests include testing for enums, permission and shown strikes.

Issue number: closes #267 

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] CHANGELOG.md has been updated. [Guide](https://tihlde.slab.com/posts/changelog-z8hybjom)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] The fixtures have been updated if needed (for migrations)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: app/content/tests/test_strike_model.py [hunk 1]
```diff
@@ -1,93 +1,126 @@
...
 
-from app.content.factories.strike_factory import StrikeFactory
+import pytest
+
+from app.content.factories import StrikeFactory
 from app.util.utils import getTimezone
 
 
+@pytest.mark.django_db
 @mock.patch("app.content.models.strike.now")
```
Comment (olros) [inline app/content/tests/test_strike_model.py:11]: image
```suggestion
@mock.patch("app.util.utils.now")
```

Det er vel egentlig teknisk sett `app.util.utils.now` som mockes?

File: app/content/tests/test_strike_model.py [hunk 2]
```diff
@@ -1,93 +1,126 @@
...
+    if created less 20 days before the winter holiday"""
+    created_at = created_at.replace(tzinfo=getTimezone())
+
+    strike = StrikeFactory.build(created_at=created_at)
 
     created_year = strike.created_at.year
     expired_year = strike.expires_at.year
 
-    assert expired_year == created_year + 1
+    assert (expired_year == created_year) == is_different_year
```
Comment (olros) [inline app/content/tests/test_strike_model.py:126]: Her sjekker du om det er samme år, altså burde `is_different_year` hete `is_same_year`

File: app/tests/content/test_strike_integration.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        (Groups.TIHLDE, status.HTTP_403_FORBIDDEN),
+    ],
+)
+def test_list_strikes_as_member_of_board_or_sub_group(
+    member, group_name, expected_status_code
+):
+    """A member of HS, Index, NOK or Sosialen can list strikes"""
+    client = get_api_client(user=member, group_name=group_name)
+    url = API_STRIKE_BASE_URL
+    response = client.get(url)
```
Comment (olros) [inline app/tests/content/test_strike_integration.py:45]: Unødvendig å deklarere url lik API_STRIKE_BASE_URL
```suggestion
    response = client.get(API_STRIKE_BASE_URL)
```

File: app/tests/content/test_strike_integration.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    """A member of HS, Index, NOK or Sosialen can create a strike"""
+    client = get_api_client(user=member, group_name=group_name)
+    response = client.post(API_STRIKE_BASE_URL, strike_post_data)
+
+    assert response.status_code == expected_status_code
+
+
+@pytest.mark.django_db
+def test_only_active_strikes_are_listed(admin_user):
+    """Out of 2 strikes created, only one active strike is shown"""
```
Comment (olros) [inline app/tests/content/test_strike_integration.py:74]:
```suggestion
    """Out of 2 strikes created, only the active strike is shown"""
```

File: app/tests/content/test_strike_integration.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    client = get_api_client(user=admin_user)
+
+    StrikeFactory.build(created_at=now() - timedelta(days=100))
```
Comment (olros) [inline app/tests/content/test_strike_integration.py:77]: Sikker på at denne vil fungere om sommeren? Hvis `now()` er 20. august så vil prikken fremdeles være aktiv fordi den fryses gjennom sommeren. Sett heller days til 365 for eks tenker jeg

File: app/tests/content/test_strike_integration.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+)
+def test_create_strikes_as_member_of_board_or_sub_group(
+    member, group_name, expected_status_code, strike_post_data
+):
+    """A member of HS, Index, NOK or Sosialen can create a strike"""
+    client = get_api_client(user=member, group_name=group_name)
+    response = client.post(API_STRIKE_BASE_URL, strike_post_data)
+
+    assert response.status_code == expected_status_code
+
```
Comment (olros) [inline app/tests/content/test_strike_integration.py:70]: Kan du legge til en test på delete-access? Kun medlemmer av HS og Index skal ha tilgang til å slette prikker

**Review by olros — CHANGES_REQUESTED** [review]
Herlig jobbet, mer tester settes alltid pris på!

File: app/content/tests/test_strike_model.py (hunk 1 above)
Comment (CJGutz) [reply app/content/tests/test_strike_model.py:11]: Da skjer det faktisk en feil. For da bruker ikke strike mocken. 
[image](https://write.agrevolution.in/python-unit-testing-mock-and-patch-8ba9c796c9c2)
 Fra: https://write.agrevolution.in/python-unit-testing-mock-and-patch-8ba9c796c9c2

File: app/content/tests/test_strike_model.py [hunk 7]
```diff
@@ -1,93 +1,126 @@
...
+        (datetime(2021, 3, 1), 20),
+        (datetime(2021, 3, 1, 1), 20),
+        (datetime(2021, 5, 9), 118),
+        (datetime(2021, 5, 9, 4), 118),
+        (datetime(2021, 5, 3), 118),
+        (datetime(2021, 8, 14), 22),
+        (datetime(2021, 11, 9), 20),
+        (datetime(2021, 11, 9, 1), 62),
+        (datetime(2021, 12, 24), 37),
+    ],
```
[the remark this answers is not in the document]
Comment (CJGutz) [reply app/content/tests/test_strike_model.py:87]: ja helt enig. Det ser bedre ut

File: app/content/tests/test_strike_model.py (hunk 2 above)
Comment (CJGutz) [reply app/content/tests/test_strike_model.py:126]: Ja ser der ja. Bra du så den

File: app/content/tests/test_strike_model.py (hunk 1 above)
Comment (olros) [reply app/content/tests/test_strike_model.py:12]: Jøss, det visste jeg ikke 👍

**Review by olros — APPROVED** [review]

**Pull request merged**